### PR TITLE
Recreate streams

### LINKED ISSUE #206: We should always recreate the configured streams
We should always recreate the configured streams if this wasn't the user's intent to remove them. 

And because Linux could change the device path, especially in a bad-contact situation, we should never trust the old device path when recreating the stream.

### LINKED ISSUE #223: We should handle device disconnection and device name change (the case of bad-contact)
Useful log section from a disconnection:

```bash
2023-04-28T14:16:33.647333Z DEBUG actix-rt|system:0|arbiter:0 ThreadId(37) HTTP request{http.method=POST http.route=/streams http.flavor=1.1 http.scheme=http http.host=0.0.0.0:6020 http.client_ip=127.0.0.1 http.user_agent=Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/112.0 http.target=/streams otel.name=HTTP POST /streams otel.kind="server" request_id=32d9c728-0329-4d01-830b-b6e086d9a9f5}:add_stream_and_start{video_and_stream_information=VideoAndStreamInformation { name: "/dev/video2 - H264 USB Camera: USB Camera", stream_information: StreamInformation { endpoints: [Url { scheme: "udp", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("0.0.0.0")), port: Some(5600), path: "", query: None, fragment: None }], configuration: Video(VideoCaptureConfiguration { encode: H264, height: 1080, width: 1920, frame_interval: FrameInterval { numerator: 1, denominator: 30 } }), extended_configuration: Some(ExtendedConfiguration { thermal: false }) }, video_source: Local(VideoSourceLocal { name: "H264 USB Camera: USB Camera", device_path: "/dev/video2", typ: Usb("usb-0000:09:00.3-3.2") }) }}:add_stream{stream=Stream { id: 77bb288d-6485-4b0b-84cb-2d78fd6774c4, pipeline: V4l(V4lPipeline { state: PipelineState { pipeline_id: 77bb288d-6485-4b0b-84cb-2d78fd6774c4, pipeline: Pipeline { inner: TypedObjectRef { inner: 0x7fb9c0089620, type: GstPipeline } }, sink_tee: Element { inner: TypedObjectRef { inner: 0x7fb9c00897e0, type: GstTee } }, sinks: {022346b5-5130-4abe-8a27-8f3df1b8c89a: Image(ImageSink { sink_id: 022346b5-5130-4abe-8a27-8f3df1b8c89a, pipeline: Pipeline { inner: TypedObjectRef { inner: 0x7fb9c019c7b0, type: GstPipeline } }, queue: Element { inner: TypedObjectRef { inner: 0x7fb9c009f020, type: GstQueue } }, proxysink: Element { inner: TypedObjectRef { inner: 0x7fb9c00e1d10, type: GstProxySink } }, _proxysrc: Element { inner: TypedObjectRef { inner: 0x7fb9c00e2410, type: GstProxySrc } }, _transcoding_elements: [Element { inner: TypedObjectRef { inner: 0x7fb9c00e7bd0, type: GstRtpH264Depay } }, Element { inner: TypedObjectRef { inner: 0x7fb9c00ead40, type: GstH264Parse } }, Element { inner: TypedObjectRef { inner: 0x7fb9c00efb80, type: GstIdentity } }, Element { inner: TypedObjectRef { inner: 0x7fb9c018e820, type: avdec_h264 } }, Element { inner: TypedObjectRef { inner: 0x7fb9c0199890, type: GstVideoConvert } }], appsink: AppSink { inner: TypedObjectRef { inner: 0x7fb9c019bdb0, type: GstAppSink } }, tee_src_pad: Some(Pad { inner: TypedObjectRef { inner: 0x7fb9c019ea70, type: GstTeePad } }), flat_samples_sender: broadcast::Sender, pad_blocker: Mutex { data: None, poisoned: false, .. }, _pipeline_runner: PipelineRunner { pipeline_weak: WeakRef(GWeakRef @ 0x7fb9c01a0c50 { priv_: GWeakRef_priv @ 0x7fb9c01a0c50 { p: 0x7fb9c019c7b0 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>), start_signal_sender: broadcast::Sender, killswitch_sender: broadcast::Sender, _killswitch_receiver: broadcast::Receiver, _watcher_thread_handle: JoinHandle { .. } }, thumbnails: Mutex { data: CachedThumbnails { map: {} }, poisoned: false, .. } }), 51c01fb8-306d-4145-9001-dc39ad25f782: Udp(UdpSink { sink_id: 51c01fb8-306d-4145-9001-dc39ad25f782, pipeline: Pipeline { inner: TypedObjectRef { inner: 0x7fb9c009a600, type: GstPipeline } }, queue: Element { inner: TypedObjectRef { inner: 0x7fb9c008ba20, type: GstQueue } }, proxysink: Element { inner: TypedObjectRef { inner: 0x7fb9c008ef80, type: GstProxySink } }, _proxysrc: Element { inner: TypedObjectRef { inner: 0x7fb9c008fc70, type: GstProxySrc } }, _udpsink: Element { inner: TypedObjectRef { inner: 0x7fb9c00980e0, type: GstMultiUDPSink } }, udpsink_sink_pad: Pad { inner: TypedObjectRef { inner: 0x7fb9c00984f0, type: GstPad } }, tee_src_pad: Some(Pad { inner: TypedObjectRef { inner: 0x7fb9c009f660, type: GstTeePad } }), addresses: [Url { scheme: "udp", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("0.0.0.0")), port: Some(5600), path: "", query: None, fragment: None }], _pipeline_runner: PipelineRunner { pipeline_weak: WeakRef(GWeakRef @ 0x7fb9c00943c0 { priv_: GWeakRef_priv @ 0x7fb9c00943c0 { p: 0x7fb9c009a600 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>), start_signal_sender: broadcast::Sender, killswitch_sender: broadcast::Sender, _killswitch_receiver: broadcast::Receiver, _watcher_thread_handle: JoinHandle { .. } } })}, pipeline_runner: PipelineRunner { pipeline_weak: WeakRef(GWeakRef @ 0x7fb9c0059f70 { priv_: GWeakRef_priv @ 0x7fb9c0059f70 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>), start_signal_sender: broadcast::Sender, killswitch_sender: broadcast::Sender, _killswitch_receiver: broadcast::Receiver, _watcher_thread_handle: JoinHandle { .. } }, _watcher_thread_handle: JoinHandle { id: Id(60) } } }), video_and_stream_information: VideoAndStreamInformation { name: "/dev/video2 - H264 USB Camera: USB Camera", stream_information: StreamInformation { endpoints: [Url { scheme: "udp", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("0.0.0.0")), port: Some(5600), path: "", query: None, fragment: None }], configuration: Video(VideoCaptureConfiguration { encode: H264, height: 1080, width: 1920, frame_interval: FrameInterval { numerator: 1, denominator: 30 } }), extended_configuration: Some(ExtendedConfiguration { thermal: false }) }, video_source: Local(VideoSourceLocal { name: "H264 USB Camera: USB Camera", device_path: "/dev/video2", typ: Usb("usb-0000:09:00.3-3.2") }) }, mavlink_camera: None }}:update_settings: src/settings/manager.rs:130: Settings saved: SettingsStruct {
    header: HeaderSettingsFile {
        name: "Camera Manager",
        version: 0,
    },
    mavlink_endpoint: None,
    streams: [
        VideoAndStreamInformation {
            name: "/dev/video2 - H264 USB Camera: USB Camera",
            stream_information: StreamInformation {
                endpoints: [
                    Url {
                        scheme: "udp",
                        cannot_be_a_base: false,
                        username: "",
                        password: None,
                        host: Some(
                            Domain(
                                "0.0.0.0",
                            ),
                        ),
                        port: Some(
                            5600,
                        ),
                        path: "",
                        query: None,
                        fragment: None,
                    },
                ],
                configuration: Video(
                    VideoCaptureConfiguration {
                        encode: H264,
                        height: 1080,
                        width: 1920,
                        frame_interval: FrameInterval {
                            numerator: 1,
                            denominator: 30,
                        },
                    },
                ),
                extended_configuration: Some(
                    ExtendedConfiguration {
                        thermal: false,
                    },
                ),
            },
            video_source: Local(
                VideoSourceLocal {
                    name: "H264 USB Camera: USB Camera",
                    device_path: "/dev/video2",
                    typ: Usb(
                        "usb-0000:09:00.3-3.2",
                    ),
                },
            ),
        },
    ],
}
2023-04-28T14:16:33.647777Z  INFO actix-rt|system:0|arbiter:0 ThreadId(37) /home/joaoantoniocardoso/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-web-4.3.1/src/middleware/logger.rs:421: 127.0.0.1 "POST /streams HTTP/1.1" 200 818 "http://0.0.0.0:6020/" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/112.0" 0.052198    
2023-04-28T14:16:33.678250Z  INFO actix-rt|system:0|arbiter:0 ThreadId(37) /home/joaoantoniocardoso/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-web-4.3.1/src/middleware/logger.rs:421: 127.0.0.1 "GET /v4l HTTP/1.1" 200 14215 "http://0.0.0.0:6020/" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/112.0" 0.027524    
2023-04-28T14:16:33.681434Z  INFO actix-rt|system:0|arbiter:0 ThreadId(37) /home/joaoantoniocardoso/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-web-4.3.1/src/middleware/logger.rs:421: 127.0.0.1 "GET /streams HTTP/1.1" 200 818 "http://0.0.0.0:6020/" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/112.0" 0.000373    
2023-04-28T14:16:33.706315Z DEBUG PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:120: Starting signal received in Pipeline 77bb288d-6485-4b0b-84cb-2d78fd6774c4
2023-04-28T14:16:34.865443Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 1
2023-04-28T14:16:34.965747Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 2
2023-04-28T14:16:35.066004Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 3
2023-04-28T14:16:35.166284Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 4
2023-04-28T14:16:35.266591Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 5
2023-04-28T14:16:35.366816Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 6
2023-04-28T14:16:35.467135Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 7
2023-04-28T14:16:35.567454Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 8
2023-04-28T14:16:35.667732Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 9
2023-04-28T14:16:35.768010Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 10
2023-04-28T14:17:13.602886Z ERROR PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:185: Error from Some("/GstPipeline:pipeline0/GstV4l2Src:v4l2src0"): Could not read from resource. (Some("../gstreamer/subprojects/gst-plugins-good/sys/v4l2/gstv4l2bufferpool.c(1198): gst_v4l2_buffer_pool_poll (): /GstPipeline:pipeline0/GstV4l2Src:v4l2src0:\npoll error 1: Resource temporarily unavailable (11)"))
2023-04-28T14:17:13.703201Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 1
2023-04-28T14:17:13.703313Z ERROR PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:185: Error from Some("/GstPipeline:pipeline0/GstV4l2Src:v4l2src0"): Failed to allocate a buffer (Some("../gstreamer/subprojects/gst-plugins-good/sys/v4l2/gstv4l2src.c(1374): gst_v4l2src_create (): /GstPipeline:pipeline0/GstV4l2Src:v4l2src0"))
2023-04-28T14:17:13.803489Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 2
2023-04-28T14:17:13.803642Z ERROR PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:185: Error from Some("/GstPipeline:pipeline0/GstV4l2Src:v4l2src0"): Internal data stream error. (Some("../gstreamer/subprojects/gstreamer/libs/gst/base/gstbasesrc.c(3132): gst_base_src_loop (): /GstPipeline:pipeline0/GstV4l2Src:v4l2src0:\nstreaming stopped, reason error (-5)"))
2023-04-28T14:17:13.903854Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:154: Position did not change 3
2023-04-28T14:17:13.903963Z  WARN PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) runner{pipeline_weak=WeakRef(GWeakRef @ 0x7fb9c0077800 { priv_: GWeakRef_priv @ 0x7fb9c0077800 { p: 0x7fb9c0089620 } }, PhantomData<*mut gstreamer::auto::pipeline::Pipeline>) pipeline_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4 killswitch_receiver=broadcast::Receiver start_signal_receiver=broadcast::Receiver allow_block=false}: src/stream/pipeline/runner.rs:181: Received EndOfStream: Eos(
    Message {
        ptr: 0x00007fb98400d2e0,
        type: "eos",
        seqnum: 58,
        src: Some(
            "pipeline0",
        ),
        structure: None,
    },
)
2023-04-28T14:17:13.904056Z  INFO PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) src/stream/pipeline/runner.rs:51: PipelineWatcher ended with no error.
2023-04-28T14:17:13.904096Z  INFO PipelineRunner-77bb288d-6485-4b0b-84cb-2d78fd6774c4 ThreadId(46) src/stream/pipeline/runner.rs:58: Error sent to killswitch channel!
2023-04-28T14:17:13.904180Z DEBUG                         actix-rt|system:0|arbiter:0 ThreadId(37) src/stream/pipeline/mod.rs:142: Killswitch received as 77bb288d-6485-4b0b-84cb-2d78fd6774c4 from PipelineState's watcher. Reason: "Normal ending"
2023-04-28T14:17:13.904415Z DEBUG                         actix-rt|system:0|arbiter:0 ThreadId(37) remove_stream{stream_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4}:update_settings: src/settings/manager.rs:130: Settings saved: SettingsStruct {
    header: HeaderSettingsFile {
        name: "Camera Manager",
        version: 0,
    },
    mavlink_endpoint: None,
    streams: [],
}
2023-04-28T14:17:13.908235Z  INFO                         actix-rt|system:0|arbiter:0 ThreadId(37) remove_stream{stream_id=77bb288d-6485-4b0b-84cb-2d78fd6774c4}: src/stream/manager.rs:547: Stream 77bb288d-6485-4b0b-84cb-2d78fd6774c4 successfully removed!
2023-04-28T14:17:13.908518Z  INFO                         actix-rt|system:0|arbiter:0 ThreadId(37) src/stream/pipeline/mod.rs:146: Pipeline 77bb288d-6485-4b0b-84cb-2d78fd6774c4 removed. Reason: Normal ending
2023-04-28T14:17:13.935171Z ERROR PipelineRunner-51c01fb8-306d-4145-9001-dc39ad25f782 ThreadId(47) src/stream/pipeline/runner.rs:48: PipelineWatcher ended with error: Failed receiving start signal: Closed
2023-04-28T14:17:13.935273Z ERROR PipelineRunner-51c01fb8-306d-4145-9001-dc39ad25f782 ThreadId(47) src/stream/pipeline/runner.rs:56: Failed to broadcast error from PipelineWatcher. Reason: channel closed
2023-04-28T14:17:13.973084Z ERROR PipelineRunner-022346b5-5130-4abe-8a27-8f3df1b8c89a ThreadId(48) src/stream/pipeline/runner.rs:48: PipelineWatcher ended with error: Failed receiving start signal: Closed
2023-04-28T14:17:13.973150Z ERROR PipelineRunner-022346b5-5130-4abe-8a27-8f3df1b8c89a ThreadId(48) src/stream/pipeline/runner.rs:56: Failed to broadcast error from PipelineWatcher. Reason: channel closed
```

Useful log section of a bad-contact:

### LINKED ISSUE #226: Inconsistent device behavior
Hard to replicate, but: 

  - `/dev/video2` exists in device tree,
  - MCM shows it as a device option, 
  - MCM fails to create the stream because it _"doesn't exist"_.

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
When a camera is disconnected during a stream, the system searches for the best device candidates that match the characteristics of the one in use on that stream, and recreates the stream from the ground up (while keeping the same pipeline ID). In the case of more than one stream running, the system discards any other device in the other stream.

To achieve that, the owner of the `<VideoAndStreamInformation>` was changed from the `<PipelineState>` (the inner state of each kind of `<Pipeline>`) to the `<Stream>`, while we moved part of the `<Stream>`'s fields to a thread-safe inner state called `<StreamState>`.

Closes #206
Closes #223
Closes #226

It's working, but it's a draft until I:
- [x] Organize the thread communication/end-of-life;
- [x] Make the recreated streams use the same ID as the killed one;
- [x] Avoid huge logs by reporting at geometrically growing intervals;

### CODE REVIEW

File: src/stream/pipeline/runner.rs [hunk 1]
```diff
@@ -115,16 +115,14 @@ impl PipelineRunner {
                                     && current_previous_position == position
                                 {
                                     lost_timestamps += 1;
-                                    warn!("Position did not change {lost_timestamps}");
```
Comment (patrickelectric) [inline src/stream/pipeline/runner.rs:154]: Is there a reason to remove this one ? IIRC that's kind of necessary to identify some cameras that can have problems

Comment (joaoantoniocardoso) [reply src/stream/pipeline/runner.rs:154]: Yeah, good one, I'm trying to remove repetitive logs that don't actually add value. It already has a log when it normalizes, I'll add a log when it starts too, then we'll have all the necessary information.

**Review by patrickelectric — APPROVED** [review]
That's a great PR, the code and organization looks awesome.

**Comment by patrickelectric** [comment]
Be free to merge once the CI approve.

**Comment by joaoantoniocardoso** [comment]
@patrickelectric added a log to record when the lost timestamps started:

https://github.com/mavlink/mavlink-camera-manager/assets/5920286/fac3b91d-27cd-4c36-970b-5af3720c86bc

**Pull request merged**